### PR TITLE
[+] Smart Search - Limit search to specific collection fields

### PR DESCRIPTION
**Pull request by ArnaudValensi**
To configure a collection:
```
return [
  'secret_key' => '7a3049f8cd14e719c219e6d75e2a7715eef38bf6ec4be2ffc774a3374c9196a8',
  'auth_key' => 'l8oeFltNrdrbq6pPeEGzxbG47h2qCK46',
  'models_path' => 'app',
  'url' => 'http://localhost:3001',
  'debug_mode' => true,
  'collection' => [
    'AppPost' => [
      'search_fields' => ['id', 'title', 'users.password'],
    ],
    'AppUser' => [
      'actions' => [
        ['name' => 'unban']
      ]
    ]
  ]
];
```

Compared to before:
```
return [
  'secret_key' => '7a3049f8cd14e719c219e6d75e2a7715eef38bf6ec4be2ffc774a3374c9196a8',
  'auth_key' => 'l8oeFltNrdrbq6pPeEGzxbG47h2qCK46',
  'models_path' => 'app',
  'url' => 'http://localhost:3001',
  'debug_mode' => true,
  'actions' => [
    'AppUser' => [
      ['name' => 'ban']
    ]
  ]
];
```

The old way is still working though.

To create a smart action, old fashion:
https://gist.github.com/arnaudbesnier/9d10918516b9c6f15ae56e76446aa9cd#file-forestsmartactionlaravel-php

Laravel 5.5+:
https://gist.github.com/ArnaudValensi/3a5e50bc50a214f631e8e6e25aedd881

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Change Log
 
 ## [Unreleased]
+### Added
+- Smart Search - Limit search to specific colelction fields.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: typo...

File: src/Http/Services/ResourcesGetter.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 use ForestAdmin\ForestLaravel\Http\Services\ConditionSetter;
 use ForestAdmin\ForestLaravel\Http\Services\SearchBuilder;
+use ForestAdmin\ForestLaravel\Logger;
```
Comment (arnaudbesnier) [inline src/Http/Services/ResourcesGetter.php]: Nope...

File: src/Model/Collection.php [hunk 3]
```diff
@@ -41,32 +41,59 @@ public function getName() {
...
-                    foreach($collectionActions as $action) {
-                        $this->actions[] = new Action($this, $action);
-                    }
+                    $actions = $collectionActions;
                 }
-            // TODO: Remove nameOld attribute once the lianas versions older than 0.1.4 are
-            //       minority.
-            } else if (array_key_exists($collectionNameOld, $actions)) {
+                // TODO: Remove nameOld attribute once the lianas versions older than 0.1.4 are
+                //       minority.
```
Comment (arnaudbesnier) [inline src/Model/Collection.php]: Why did you change the indentation? The TODO is about the `else if`.

File: src/Http/Services/SearchBuilder.php [hunk 4]
```diff
@@ -18,27 +20,65 @@ public function __construct($query, $collectionSchema, $tableNameModel,
...
+                              $field->getField(), intval($this->params->search));
+                        } else if ($field->getType() === 'String') {
+                            $this->query->orWhereRaw('LOWER('.$s.
+                              $this->tableNameModel.$s.'.'.$s.$field->getField().
+                              $s.') LIKE LOWER(\'%'.$this->params->search.'%\')');
+                        }
                     }
                 } else if ($field->isTypeToOne() &&
                   (int)$this->params->searchExtended) {
+
```
Comment (arnaudbesnier) [inline src/Http/Services/SearchBuilder.php]: Why a newline in this case?

**Comment by ArnaudValensi** [comment]
[(5) Custom Search - Search requests take a very long time to be done on his collections and it is a blocker to go PRO customer](https://trello.com/c/2bRnlYfV/2480-5-custom-search-search-requests-take-a-very-long-time-to-be-done-on-his-collections-and-it-is-a-blocker-to-go-pro-customer)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,9 @@
 # Change Log
 
 ## [Unreleased]
+### Added
+- Smart Search - Limit search to specific collection fields.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: That is not the only thing you did in this PR (you should talk about the new Smart Actions "API").
